libmamba/src/specs/match_spec.cpp

```diff
@@ -505,9 +505,9 @@ namespace mamba::specs
     auto MatchSpec::parse(std::string_view str) -> expected_parse_t<MatchSpec>
     {
         // Remove comments, i.e. everything after ` #` (space included)
-        if (const auto idx = str.find('#'); idx != std::string::npos && str[idx - 1] == ' ')
+        if (const auto idx = str.find(" #"); idx != std::string::npos)
         {
-            str = str.substr(0, idx);
+            str = str.substr(0, idx + 1);
```

Comment:
The space can also be omitted:
```suggestion
            str = str.substr(0, idx);
```

Reply:
Sounds good, done.